Script/update_grovepi.sh

```diff
@@ -5,34 +5,147 @@ PIHOME=/home/pi
 else
-	echo "Cloning"
-    cd $PIHOME/$DEXTER/
-    git clone https://github.com/DexterInd/GrovePi
-    cd GrovePi
+  echo "Welcome to GrovePi Installer."
+fi
+echo "Updating GrovePi for $selectedbranch branch with the following options:"
+([[ $installdependencies = "true" ]] && echo "  --no-dependencies=false") || echo "  --no-dependencies=true"
+([[ $updaterepo = "true" ]] && echo "  --no-update-aptget=false") || echo "  --no-update-aptget=true"
+([[ $install_pkg_scriptools = "true" ]] && echo "  --bypass-pkg-scriptools=false") || echo "  --bypass-pkg-scriptools=true"
+echo "  --user-local=$userlocal"
+echo "  --env-local=$envlocal"
+echo "  --system-wide=$systemwide"
+
+# in case the following packages are not installed and `--no-dependencies` option has been used
+if [[ $installdependencies = "false" ]]; then
+  command -v git >/dev/null 2>&1 || { echo "I require git but it's not installed. Don't use --no-dependencies option. Aborting." >&2; exit 1; }
+  command -v python >/dev/null 2>&1 || { echo "Executable \"python\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 2; }
+  command -v python3 >/dev/null 2>&1 || { echo "Executable \"python3\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 3; }
+  command -v pip >/dev/null 2>&1 || { echo "Executable \"pip\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 4; }
+  command -v pip3 >/dev/null 2>&1 || { echo "Executable \"pip3\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 5; }
 fi
 
-change_branch $BRANCH   # Change to a branch we're working on in the GrovePi Directory. 
-                        # Variable $BRANCH comes from /upd_script/fetch.sh
+# create rest of list of arguments for script_tools call
+optionslist+=("$selectedbranch")
+[[ $usepython3exec = "true" ]] && optionslist+=("--use-python3-exe-too")
+[[ $updaterepo = "true" ]] && optionslist+=("--update-aptget")
+[[ $installdependencies = "true" ]] && optionslist+=("--install-deb-deps")
+[[ $install_pkg_scriptools = "true" ]] && optionslist+=("--install-python-package")
+
+echo "Options used for script_tools script: \"${optionslist[@]}\""
+
+# update script_tools first
+curl --silent -kL dexterindustries.com/update_tools > $PIHOME/tmp_script_tools.sh
+echo "Installing script_tools. This might take a while.."
+bash $PIHOME/tmp_script_tools.sh ${optionslist[@]} > /dev/null
+rm $PIHOME/tmp_script_tools.sh
 
-feedback "--> Start GrovePi update install."
-feedback "---------------------------------"
-pushd $PIHOME/$DEXTER/GrovePi/Script > /dev/null
+# HAVE TO UNCOMMENT ONCE check_internet makes it to script_tools
+# check if there's internet access,
+# otherwise straight out exit the script
+# check_internet
+
+# needs to be sourced from here when we call this as a standalone
+source /home/pi/$DEXTER/lib/$DEXTER/script_tools/functions_library.sh
+
+# create folders recursively if they don't exist already
+mkdir -p $DEXTER_PATH
```

Comment:
oh, I had never noticed that -p option before. Nice to have!

Reply:
Yeah, it trims down the code a lot. 😀